framework_tvb/tvb/core/adapters/abcuploader.py

```diff
@@ -77,8 +95,55 @@ def _prelaunch(self, operation, uid=None, available_disk_space=0, view_model=Non
         self.meta_data.update({DataTypeMetaData.KEY_SUBJECT: view_model.data_subject})
         self.generic_attributes.subject = view_model.data_subject
 
+        trait_upload_field_names = list(self.get_form_class().get_upload_information().keys())
+        for upload_field_name in trait_upload_field_names:
+            self._decrypt_content(view_model, upload_field_name)
+
         return ABCSynchronous._prelaunch(self, operation, uid, available_disk_space, view_model, **kwargs)
 
+    @staticmethod
+    def _decrypt_content(view_model, trait_upload_field_name):
+        if view_model.encrypted_aes_key is None:
+            return
+
+        if TvbProfile.current.UPLOAD_KEY_PATH is None or not os.path.exists(TvbProfile.current.UPLOAD_KEY_PATH):
+            raise LaunchException("We can not process Encrypted files at this moment, "
+                                  "due to missing PK for decryption! Please contact the administrator!")
+
+        upload_path = getattr(view_model, trait_upload_field_name)
+
+        # Get the encrypted password
+        with open(view_model.encrypted_aes_key, 'rb') as f:
+            encrypted_password = f.read()
+
+        # Read the private key
+        with open(TvbProfile.current.UPLOAD_KEY_PATH, "rb") as key_file:
+            private_key = serialization.load_pem_private_key(
+                key_file.read(),
+                password=None,
+                backend=default_backend()
+            )
+
+        # Decrypt the password using the private key
+        decrypted_password = private_key.decrypt(
+            encrypted_password,
+            padding.OAEP(
+                mgf=padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            )
+        )
+
+        decrypted_password = decrypted_password.decode()
+
+        # Get path to decrypted file
+        decrypted_download_path = upload_path.replace(ENCRYPTED_DATA_SUFFIX, DECRYPTED_DATA_SUFFIX)
+
+        # Use the decrypted password to decrypt the message
+        pyAesCrypt.decryptFile(upload_path, decrypted_download_path, decrypted_password,
+                               TvbProfile.current.hpc.CRYPT_BUFFER_SIZE)
```

Comment:
Could it be worthy to create a dedicated `CryptographySettings` class and the put all the constants like `CRYPT_BUFFER_SIZE, CRYPT_PASS_SIZE, ENCRYPTED_DATA_SUFFIX, DECRYPTED_DATA_SUFFIX` there ?
Or under `tvb.core.adapters.constants` ? (I have a tendency for the second alternative, as it is closer to the usages, but I could be ok with these living under the profile, also.

Reply:
I can move the constants defined by me, but CRYPT_BUFFER_SIZE AND CRYPT_PASS_SIZE I think should stay under the profile because they use a manager object given as a parameter to the constructor.